packages/react-router/tests/loaders.test.tsx

```diff
@@ -729,3 +730,89 @@ test('clears pendingTimeout when match resolves', async () => {
   expect(nestedPendingComponentOnMountMock).not.toHaveBeenCalled()
   expect(fooPendingComponentOnMountMock).not.toHaveBeenCalled()
 })
+
+test('reproducer for #4696', async () => {
+  const rootRoute = createRootRoute({
+    beforeLoad: async ({ context }) => {
+      return {
+        ...context,
+        isAuthenticated: true,
+        isAdmin: false,
+      }
+    },
+    loader: ({ context }) => context,
+    pendingComponent: () => 'Loading...',
+    wrapInSuspense: true,
+    component: RootRouteContent,
+  })
+
+  function RootRouteContent() {
+    const routeData = rootRoute.useLoaderData()
+    const isAuthenticated = routeData.isAuthenticated
+    return (
+      <>
+        {!!isAuthenticated && (
+          <nav style={{ display: 'flex', gap: 8 }}>
+            <Link to="/">Index</Link>
+            <Link to="/dashboard">Dashboard</Link>
+          </nav>
+        )}
+        <hr />
+        <Outlet />
+      </>
+    )
+  }
+
+  const indexRoute = createRoute({
+    getParentRoute: () => rootRoute,
+    path: '/',
+    component: () => <h1>Index</h1>,
+  })
+
+  const dashboardRoute = createRoute({
+    getParentRoute: () => rootRoute,
+    path: '/dashboard',
+    pendingComponent: () => 'Loading dashboard...',
+    validateSearch: () => {
+      return {
+        page: 0,
+      }
+    },
+    component: () => <h1>Dashboard</h1>,
+  })
+
+  const routeTree = rootRoute.addChildren([indexRoute, dashboardRoute])
+  const router = createRouter({ routeTree })
+
+  render(<RouterProvider router={router} />)
+  await act(async () => {
```

Comment:
replace 

```tsx
await act(async () => {

    /* no-op */
  })
  ```
  
  by
  
  ```tsx
  await router.load()
```

Reply:
for some reason when I change it to `await router.load()` for the index route, the test fails there – it doesn't render the navbar as it did with the `act` call.

<img width="979" height="456" alt="image" src="https://github.com/user-attachments/assets/1e5b6b3b-b317-460d-a704-99bf8e286cc4" />

when I use `await router.load()` for the dashboard route part, it doesn't render anything at all:

<img width="992" height="547" alt="image" src="https://github.com/user-attachments/assets/3710715d-8b2e-436d-ae85-931c97c9d0e5" />